tests/unit/subscriptions/test-class-wc-payments-subscription-change-payment-method.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * Class WC_Payments_Invoice_Service_Test
+ *
+ * @package WooCommerce\Payments\Tests
+ */
+
+use PHPUnit\Framework\MockObject\MockObject;
+use WCPay\Exceptions\API_Exception;
+
+/**
+ * WC_Payments_Invoice_Service_Test unit tests.
+ */
+class WC_Payments_Subscription_Change_Payment_Method_Test extends WP_UnitTestCase {
+
+	/**
+	 * System under test.
+	 *
+	 * @var WC_Payments_Product_Service
+	 */
+	private $change_payment_method_handler;
+
+	/**
+	 * Pre-test setup
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->change_payment_method_handler = new WC_Payments_Subscription_Change_Payment_Method_Handler();
+	}
+
+	/**
+	 * Tests for WC_Payments_Subscription_Change_Payment_Method_Handler::update_subscription_change_payment_button().
+	 */
+	public function test_update_subscription_change_payment_button() {
+		$mock_subscription = new WC_Subscription();
+
+		// Test the false case - no change to input.
+		$this->assertEquals( [], $this->change_payment_method_handler->update_subscription_change_payment_button( [], $mock_subscription ) );
+
+		// Set up a subscriotion with a failed last payment to test the positive/true case.
+		$mock_subscription->status     = 'on-hold';
+		$mock_subscription->last_order = WC_Helper_Order::create_order();
+		$mock_subscription->last_order->set_status( 'failed' );
+
+		// The update_subscription_change_payment_button function attempts to call WC_Subscription->get_checkout_payment_url(). To avoid errors mock that function's return.
+		$mock_subscription->checkout_payment_url = 'example.com';
+		$mock_subscription->update_meta_data( WC_Payments_Subscription_Service::SUBSCRIPTION_ID_META_KEY, 'sub_test123' );
+		$mock_subscription->save();
+
+		$default_actions = [
+			'suspend'               => [
+				'name' => 'suspend',
+				'url'  => 'example.com',
+			],
+			'change_payment_method' => [
+				'name' => 'Change payment',
+				'url'  => 'example.com',
+			],
+		];
+
+		$result = $this->change_payment_method_handler->update_subscription_change_payment_button( $default_actions, $mock_subscription );
+
+		$this->assertTrue( is_array( $result ) );
```

Comment:
```suggestion
		$this->assertIsArray( $result );
```

Reply:
It turns out `assertIsArray()` isn't available on at least one of the PHPUnit versions we're using. https://github.com/Automattic/woocommerce-payments/pull/2788/checks?check_run_id=3481575015

How would you usually get around that? Go back to `assertTrue`, or something else like a version check etc?

Reply:
Ah, good point! Not much we can do there unless we upgrade PHPUnit across the board, let's stick with `assertTrue` for now 👍